instaclustr/resource_cluster.go

```diff
@@ -325,29 +365,165 @@ func resourceClusterCreate(d *schema.ResourceData, meta interface{}) error {
+			cluster, err := client.ReadCluster(id)
+
+			if err != nil {
+				return resource.NonRetryableError(fmt.Errorf("[Error] Error retrieving cluster info: %s", err))
+			}
+
+			if cluster.ClusterStatus == waitForClusterState {
+
+				if isKafkaCluster {
+
+					if hasRestProxy {
+						// preparing the kafka rest proxy bundle user update request
+						updateRestProxyData := UpdateBundleUserRequest{
+							Username: "ickafkarest",
+							Password: d.Get("kafka_rest_proxy_user_password").(string),
+						}
+						var jsonStrUpdateRestProxy []byte
+						jsonStrUpdateRestProxy, err := json.Marshal(updateRestProxyData)
+
+						if err != nil {
+							return resource.NonRetryableError(fmt.Errorf("[Error] Error creating the kafka rest proxy bundle user update request : %s", err))
+						}
+
+						//updating the kafka rest proxy bundle user
+						if len(kafkaRestProxyUserPassword) > 0 {
+							err = client.UpdateBundleUser(d.Get("cluster_id").(string), "kafka_rest_proxy", jsonStrUpdateRestProxy)
+						}
+						if err != nil {
+							return resource.NonRetryableError(fmt.Errorf("[Error] Error updating the kafka rest proxy bundle user password : %s", err))
+						}
+					}
+
+					if hasSchemaRegistry {
+						// preparing the kafka schema registry bundle user update request
+						updateSchemaRegistryData := UpdateBundleUserRequest{
+							Username: "ickafkaschema",
+							Password: d.Get("kafka_schema_registry_user_password").(string),
+						}
+						var jsonStrUpdateSchemaRegistry []byte
+						jsonStrUpdateSchemaRegistry, err = json.Marshal(updateSchemaRegistryData)
+
+						if err != nil {
+							return resource.NonRetryableError(fmt.Errorf("[Error] Error creating the kafka schema registry bundle user update request : %s", err))
+						}
+
+						//updating the kafka schema registry bundle user
+						if len(kafkaSchemaRegistryUserPassword) > 0 {
+							err = client.UpdateBundleUser(d.Get("cluster_id").(string), "kafka_schema_registry", jsonStrUpdateSchemaRegistry)
+						}
+						if err != nil {
+							return resource.NonRetryableError(fmt.Errorf("[Error] Error updating the kafka schema registry bundle user password : %s", err))
+						}
+					}
+
+					return resource.NonRetryableError(resourceClusterRead(d, meta))
+				} else if !isKafkaCluster && (len(kafkaSchemaRegistryUserPassword) > 0 || len(kafkaRestProxyUserPassword) > 0) {
+					return resource.NonRetryableError(fmt.Errorf("[Error] Error updating the bundle user passwords, because it should be a KAFKA cluster in order to update the schema-registry or rest-proxy users"))
+				}
+			}
+
+			return resource.RetryableError(fmt.Errorf("[Error] Cluster is in state %s", cluster.ClusterStatus))
+		})
+	} else {
+		return resourceClusterRead(d, meta)
+	}
 }
 
 func resourceClusterUpdate(d *schema.ResourceData, meta interface{}) error {
 	d.Partial(true)
-	// currently only cluster resize is supported
-	if !d.HasChange("node_size") {
```

Comment:
We should be either be throwing an error for all the properties that we don't support update for, or add ForceNew: true to all those non-updatable properties (ForceNew is the preferable way). Otherwise, this can lead to issues where after updating a non-updatable property, `terraform plan` will always suggest that there is a change to the resource.